### ISSUE TITLE
Retransmit when server is unreachable

### ISSUE THREAD

**Issue by al1img** [obgm/libcoap#418]
libcoap release-4.2.0:

Simple client on linux sends GET message to the server. The server is down. After first attempt the client gets ICMP message: destination unreachable and disconnects the session. At next attempts the client doesn't send any message, just waits retransmission timeout. 

```
Oct 31 20:04:39 DEBG ***[::1]:56216 <-> [::1]:5683 UDP : new outgoing session
Oct 31 20:04:39 DEBG *  [::1]:56216 <-> [::1]:5683 UDP : sent 83 bytes
Oct 31 20:04:39 DEBG ** [::1]:56216 <-> [::1]:5683 UDP : tid=29546 added to retransmit queue (2313ms)
Oct 31 20:04:39 WARN coap_network_read: unreachable
Oct 31 20:04:42 DEBG ** [::1]:56216 <-> [::1]:5683 UDP : tid=29546: retransmission #1
Oct 31 20:04:47 DEBG ** [::1]:56216 <-> [::1]:5683 UDP : tid=29546: retransmission #2
Oct 31 20:04:57 DEBG ** [::1]:56216 <-> [::1]:5683 UDP : tid=29546: retransmission #3
Oct 31 20:05:16 DEBG ** [::1]:56216 <-> [::1]:5683 UDP : tid=29546: retransmission #4
Oct 31 20:05:54 DEBG ** [::1]:56216 <-> [::1]:5683 UDP : tid=29546: give up after 4 attempts
```

Issue 1: It is not obvious that the session is closed at this case. And for new transmit the session should be reopened. Is there any callback to notify the client that the session is closed?

Issue 2: If the server becomes available between next retransmissions (for example between `#2`and `#3`), the message is failed because the session is closed and attempt `#3` and following don't actually send pdu.

**Comment by mrdeep1** [comment]
@al1img This should be fixed by #356 (for the PDU re-transmission) and #357 (for the callback), both of which are in the develop branch.

Please check with the develop version that this is no longer an issue for you.

**Comment by al1img** [comment]
@mrdeep1 Thanks for pointing me out. I confirm that the issues are resolved in the develop branch.
The issue can be closed.
